retrofit/src/main/java/retrofit2/ParameterHandler.java

```diff
@@ -174,32 +176,69 @@ void apply(RequestBuilder builder, @Nullable Map<String, T> value) throws IOExce
         throw Utils.parameterError(method, p, "Query map was null");
       }
 
+      boolean fail = false;
+      String entryKey = null;
+      T entryValue = null;
       for (Map.Entry<String, T> entry : value.entrySet()) {
-        String entryKey = entry.getKey();
+        entryKey = entry.getKey();
         if (entryKey == null) {
           throw Utils.parameterError(method, p, "Query map contained null key.");
         }
-        T entryValue = entry.getValue();
+        entryValue = entry.getValue();
         if (entryValue == null) {
           throw Utils.parameterError(
               method, p, "Query map contained null value for key '" + entryKey + "'.");
         }
-
-        String convertedEntryValue = valueConverter.convert(entryValue);
-        if (convertedEntryValue == null) {
-          throw Utils.parameterError(
-              method,
-              p,
-              "Query map value '"
-                  + entryValue
-                  + "' converted to null by "
-                  + valueConverter.getClass().getName()
-                  + " for key '"
-                  + entryKey
-                  + "'.");
+        String convertedEntryValue;
+        if (entryValue instanceof List || entryValue.getClass().isArray()) {
```

Comment:
multi-value map `get` method return value not only a `List`,it also can be a `Collection`, so use `Iterable ` is more suitable?